airflow/utils/log/file_task_handler.py

```diff
@@ -182,14 +184,38 @@ def set_context(self, ti: TaskInstance) -> None | SetContextPropagate:
         functionality is only used in unit testing.
 
         :param ti: task instance object
+        :param identifier: if set, adds suffix to log file. For use when shipping exceptional messages
+            to task logs from context other than task or trigger run
         """
-        local_loc = self._init_file(ti)
+        local_loc = self._init_file(ti, identifier=identifier)
         self.handler = NonCachingFileHandler(local_loc, encoding="utf-8")
         if self.formatter:
             self.handler.setFormatter(self.formatter)
         self.handler.setLevel(self.level)
         return SetContextPropagate.MAINTAIN_PROPAGATE if self.maintain_propagate else None
 
+    @cached_property
+    def _supports_arbitrary_ship(self):
+        return "identifier" in inspect.signature(self.set_context).parameters
+
+    def ship_arbitrary_message(
+        self, *, ti: TaskInstanceKey | TaskInstance, identifier: str, message: str, level: int
+    ):
+        """For sending exception messages to task logs visible in UI."""
+        if not self._supports_arbitrary_ship:
+            return
+        h = copy(self)
+        try:
+            if hasattr(h, "mark_end_on_close"):
+                h.mark_end_on_close = False
+            h.set_context(ti, identifier=identifier)
+            filename, lineno, func, stackinfo = logger.findCaller()
+            record = logging.LogRecord("", level, filename, lineno, message, None, None, func=func)
```

Comment:
it does feel a bit hacky to build the logrecord in this way... is it necessary to do ? i don't recall

Reply:
added component for name attribute here as discussed